test/integration/features/profile.feature

```diff
@@ -2,104 +2,251 @@
+    Then stdout should match "-\s*foo\s*Does Not Exist\s*\(Active\)"
+     And stdout should match "-\s*minishift\s*Does Not Exist"
+     And stdout should match "-\s*123foo\s*Does Not Exist"
+     And stdout should match "-\s*test-123\s*Does Not Exist"
+     And stdout should match "-\s*profile_\s*Does Not Exist"
+     And stdout should match "-\s*foo-\s*Does Not Exist"
+     And stdout should match "-\s*vm_45\s*Does Not Exist"
+     And stdout should match "-\s*20\s*Does Not Exist"
+     And stdout should match "-\s*random001z\s*Does Not Exist"
+     And stdout should match "-\s*00_XYZ-\s*Does Not Exist"
 
-  Scenario: User should be able to list 'foo' and 'minishift'
-      When executing "minishift profile list" succeeds
+  Scenario: As user, I can switch between existing profiles
+    Given Profile "123foo" has state "Does Not Exist"
+     When executing "minishift profile set 123foo" succeeds
       Then stdout should contain
       """
-      - foo		Running		(Active)
-      - minishift	Running
+      Profile '123foo' set as active profile
       """
-
-  Scenario: User should be able set 'minishift' as the active profile
-      When executing "minishift profile set minishift" succeeds
+     When executing "minishift profile set minishift" succeeds
       Then stdout should contain
       """
       Profile 'minishift' set as active profile
-      """
+      """   
 
-Scenario: User should be able to list 'minishift' as the active profile
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
-    """
-    - minishift	Running		(Active)
-    """
-
-Scenario: User should be able to delete profile 'foo'
-    When executing "minishift profile delete foo --force" succeeds
-    Then stdout should contain
-    """
-    Profile 'foo' deleted successfully
-    """
-
-Scenario: User can not delete default profile 'minishift'
-    When executing "minishift profile delete minishift --force"
-    Then exitcode should equal "1"
-     And stderr should contain
+  Scenario: As user, I can start Minishift with default profile 'minishift'
+    Given Profile "minishift" has state "Does Not Exist"
+     And executing "minishift profile set minishift" succeeds
+     Then stdout should contain
      """
-     Default profile 'minishift' can not be deleted
+     Profile 'minishift' set as active profile
      """
+     When executing "minishift start" succeeds
+      Then Profile "minishift" should have state "Running"
+       And executing "minishift profile list" succeeds
+       And stdout should match "-\s*minishift\s*Running\s*\(Active\)"
 
-Scenario: Deleting Minishift
-   Given Minishift has state "Running"
-    When executing "minishift delete --force" succeeds
-    Then Minishift should have state "Does Not Exist"
-    When executing "minishift ip"
-    Then exitcode should equal "1"
+  Scenario: As user, I can get internal docker registry address of default profile 'minishift'
```

Comment:
This really does not belong here. Why is the internal docker registry specific to the profile feature?

Reply:
Yup..this should be moved. Better place for this scenario would be, if we add it to issue like #1318 where the registry validation will be done.